app/pages/Account/functions.js

```diff
@@ -0,0 +1,313 @@
+        // This reveal is for the current bid => Bid is revealed
+        isRevealed = true;
+        revealPrevOut = { hash: reveal.prevout.hash, index: inputIdx };
+        break;
+      }
+    }
+
+    // If bid not yet revealed and is in reveal period
+    if (!isRevealed) {
+      if (auction.state === "REVEAL") {
+        revealableHNS += blind;
+        revealableNum++;
+        if (
+          revealableBlock === null ||
+          auction.stats.blocksUntilClose < revealableBlock
+        ) {
+          revealableBlock = auction.stats.blocksUntilClose;
+        }
+      }
+      continue;
+    }
+
+    // At this point, the bid is revealed (at least)
+
+    // If still in reveal period, the blind is no longer locked
+    if (auction.state === "REVEAL") {
+      revealingHNS -= blind;
+      continue;
+    }
+
+    // Get the reveal coin from wallet
+    const revealCoin = await walletClient.getCoin(
+      revealPrevOut.hash,
+      revealPrevOut.index
+    );
+    console.log("revealCoin", auction.name, bid.value, revealCoin);
+
+    // Check if the reveal coin is the owner of the name
+    const revealCoinIsOwner =
+      auction.owner.hash === revealPrevOut.hash &&
+      auction.owner.index === revealPrevOut.index;
+    console.log("revealCoinIsOwner", revealCoinIsOwner);
+
+    // If the reveal coin is the owner, then the name is registerable
+    if (revealCoinIsOwner) {
+      registerableHNS += auction.highest - auction.value;
+      registerableNum++;
+      continue;
+    }
+
+    // If reveal coin exists, but not owner, then it is redeemable
+    if (revealCoin) {
+      // Is local?
+      if (revealCoin.height < auction.height) {
+        continue;
+      }
+      redeemableHNS += bid.value;
+      redeemableNum++;
+    }
+  }
+
+  // locked Bidding = bids in BIDDING state auctions
+  // locked Reveal = bids in REVEAL state auctions (includes blind if not revealed yet)
+  // locked Finished = bids that can be REDEEMed or REGISTERed
+  return {
+    lockedBalance: {
+      bidding: { HNS: biddingHNS, num: biddingNum },
+      revealing: { HNS: revealingHNS, num: revealingNum },
+      finished: {
+        HNS: registerableHNS + redeemableHNS,
+        num: registerableNum + redeemableNum,
+      },
+    },
+    actionableInfoBids: {
+      revealable: {
+        HNS: revealableHNS,
+        num: revealableNum,
+        block: revealableBlock,
+      },
+      redeemable: { HNS: redeemableHNS, num: redeemableNum },
+      registerable: { HNS: registerableHNS, num: registerableNum },
+    },
+  };
+}
+
+async function getStatsFromNames(height, network) {
+  let transferringDomains = new Set();
+  let transferringBlock = null;
+  let finalizableDomains = new Set();
+  let renewableDomains = new Set();
+  let renewableBlock = null;
+
+  const names = await walletClient.getNames();
+  console.log("names", names);
+
+  for (let name of names) {
+    const auction = await walletClient.getAuctionInfo(name.name);
```

Comment:
I would call the following directly instead of `walletClient.getAuctionInfo`, since each rpc call is an http request that has overhead. If a wallet has thousands of name, it might be a bit slow.

```
const ns = await wallet.getNameStateByName(name);
json = ns?.getJSON(this.lastKnownChainHeight, this.network);
```